druid/examples/multiwin.rs

```diff
@@ -79,7 +81,58 @@ fn ui_builder() -> impl Widget<State> {
     row.add_child(Padding::new(5.0, inc_button));
     row.add_child(Padding::new(5.0, dec_button));
     col.add_flex_child(Align::centered(row), 1.0);
-    col
+    Glow::new(col)
+}
+
+struct Glow<W> {
+    inner: WidgetPod<State, W>,
+}
+
+impl<W: Widget<State>> Glow<W> {
+    pub fn new(inner: W) -> Glow<W> {
+        Glow {
+            inner: WidgetPod::new(inner),
+        }
+    }
+}
+
+impl<W: Widget<State>> Widget<State> for Glow<W> {
+    fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut State, env: &Env) {
+        self.inner.event(ctx, event, data, env);
+    }
+
+    fn lifecycle(&mut self, ctx: &mut LifeCycleCtx, event: &LifeCycle, data: &State, env: &Env) {
+        if let LifeCycle::HotChanged(_) = event {
+            ctx.request_paint();
+        }
+        self.inner.lifecycle(ctx, event, data, env);
+    }
+
+    fn update(&mut self, ctx: &mut UpdateCtx, old_data: &State, data: &State, env: &Env) {
+        if old_data.glow_hot != data.glow_hot {
+            ctx.request_paint();
+        }
+        self.inner.update(ctx, data, env);
+    }
+
+    fn layout(
+        &mut self,
+        ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        data: &State,
+        env: &Env,
+    ) -> Size {
+        let size = self.inner.layout(ctx, bc, data, env);
+        self.inner.set_layout_rect(ctx, data, env, size.to_rect());
+        size
```

Comment:
a general note; I think it's good to call `bc.constrain(size)` anytime you return from layout, as a way of ensuring all returned sizes are appropriate for the constraints.

Reply:
Well, my inspiration came from [`ControllerHost`](https://github.com/xi-editor/druid/blob/master/druid/src/widget/controller.rs) which does not do that, maybe open an Issue to make this consistent for all container?